### DISCUSSION TITLE
计算指定值区间时长

### DISCUSSION THREAD

**Discussion by axinger** [apache/iotdb#5893] [Q&A]
怎么统计： 
 开机-关机，  开机-关机， 开机- 
onlin 值为  true 至false 以及 true 至 现在 , 这样区间,设备在线时长?
<img width="424" alt="图片" src="https://user-images.githubusercontent.com/18339649/168197654-26d66d99-d707-4963-93ae-8da0f2a5538f.png">

**Comment by jixuan1989** [comment]
now we may need to use UDF. 
Summer 2022 has a related feature, see https://summer-ospp.ac.cn/#/org/prodetail/22ecc0278
